lua/nvim-treesitter/configs.lua

```diff
@@ -242,81 +35,89 @@ local config = {
 
 local M = {}
 
-local function enable_module(mod, bufnr, ft)
+local function enable_module(mod, bufnr, lang)
   local bufnr = bufnr or api.nvim_get_current_buf()
-  local ft = ft or api.nvim_buf_get_option(bufnr, 'ft')
-  if not parserlist[ft] or not config.modules[mod] then
+  local lang = lang or parsers.ft_to_lang(api.nvim_buf_get_option(bufnr, 'ft'))
+  if not parsers.list[lang] or not config.modules[mod] then
     return
   end
 
   local loaded_mod = require(string.format("nvim-treesitter.%s", mod))
-  loaded_mod.attach(bufnr, ft)
+  loaded_mod.attach(bufnr, lang)
 end
 
-local function enable_mod_conf_autocmd(mod, ft)
-  if not config.modules[mod] or M.is_enabled(mod, ft) then return end
+local function enable_mod_conf_autocmd(mod, lang)
+  if not config.modules[mod] or M.is_enabled(mod, lang) then return end
 
+  local ft = parsers.lang_to_ft(lang)
   local cmd = string.format("lua require'nvim-treesitter.%s'.attach()", mod)
   api.nvim_command(string.format("autocmd FileType %s %s", ft, cmd))
   for i, parser in pairs(config.modules[mod].disable) do
-    if parser == ft then
+    if parser == lang then
       table.remove(config.modules[mod].disable, i)
       break
     end
   end
 end
 
-local function enable_all(mod, ft)
+local function enable_all(mod, lang)
   if not config.modules[mod] then return end
 
   for _, bufnr in pairs(api.nvim_list_bufs()) do
-    if not ft or api.nvim_buf_get_option(bufnr, 'ft') == ft then
-      enable_module(mod, bufnr, ft)
+    local ft = api.nvim_buf_get_option(bufnr, 'ft')
+    if not lang or ft == parsers.lang_to_ft(lang) then
+      enable_module(mod, bufnr, lang)
     end
   end
-  if ft then
-    if parsers.has_parser(ft) then
-      enable_mod_conf_autocmd(mod, ft)
+  if lang then
+    if parsers.has_parser(lang) then
+      enable_mod_conf_autocmd(mod, lang)
     end
   else
-    for _, ft in pairs(M.available_parsers()) do
-      if parsers.has_parser(ft) then
-        enable_mod_conf_autocmd(mod, ft)
+    for _, lang in pairs(parsers.available_parsers()) do
+      if parsers.has_parser(lang) then
+        enable_mod_conf_autocmd(mod, lang)
       end
     end
   end
   config.modules[mod].enable = true
 end
 
-local function disable_module(mod, bufnr, ft)
+local function disable_module(mod, bufnr, lang)
   local bufnr = bufnr or api.nvim_get_current_buf()
-  local ft = ft or api.nvim_buf_get_option(bufnr, 'ft')
-  if not parserlist[ft] or not config.modules[mod] then
+  local lang = lang or parsers.ft_to_lang(api.nvim_buf_get_option(bufnr, 'ft'))
+  if not lang then
+    return
+  end
+
+  if not parsers.list[lang] or not config.modules[mod] then
     return
   end
 
   local loaded_mod = require(string.format("nvim-treesitter.%s", mod))
   loaded_mod.detach(bufnr, ft)
```

Comment:
@kyazdani42 there are still a few occurrences of  'ft' left! ft is undefined here.

Reply:
Also in `M.is_enabled`

Reply:
Also `locals.lua` needs to be updated (ft -> lang). Just seeing this now when playing a bit with nvim-treesitter.

Reply:
woops sorry guys i thought you reviewed the code before that !
I will push an update thanks for notifying

Reply:
Reviewing was looking at the code and test some basic functionality. Disabling is not triggered that often. But the linters see it. With proper CI this will not happen in future.